### PR TITLE
Purchases: Update me/purchases module to ECMAScript 6

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request updates the `client/me/purchases` section to use as many [ES6](https://github.com/Automattic/calypso-pre-oss/blob/master/docs/coding-guidelines/javascript.md#es6) features as possible. It also fixes and improves a few minor things in the process - such as missing props validation.
 
screenshot

#### Additional notes
 
You should probably consider browsing commits [one by one](https://github.com/Automattic/wp-calypso/pull/4346/commits) to review this pull request since it might be easier to understand the changes rather than in the [unified diff view](https://github.com/Automattic/wp-calypso/pull/4346/files).

#### Testing instructions
 
1. Run `git checkout update/me-purchases-to-es6` and start your server
2. Purchase different types of products, using different methods of payment
3. Check that the [`Purchases` section](http://calypso.localhost:3000/purchases) behave as before

#### Reviews
 
- [x] Code
- [x] Product

### CODE REVIEW

File: client/me/purchases/cancel-purchase/loading-placeholder.jsx [hunk 1]
```diff
@@ -6,39 +6,35 @@ import React from 'react';
...
-				<CompactCard>
-					<Button className="cancel-purchase-loading-placeholder__cancel-button" />
-				</CompactCard>
-			</LoadingPlaceholder>
-		);
-	}
-} );
+CancelPurchaseLoadingPlaceholder.propTypes = {
+	purchaseId: React.PropTypes.string.isRequired,
+	selectedSite: React.PropTypes.object.isRequired
```
Comment (drewblaisdell) [inline client/me/purchases/cancel-purchase/loading-placeholder.jsx]: If local storage is empty when we mount `/purchases/:site/:purchaseId/cancel`, `selectedSite` is false, and the page fails to load with [errors in the console](https://cloudup.com/c0Hx4Vk6Htr).

File: client/me/purchases/confirm-cancel-purchase/loading-placeholder.jsx [hunk 2]
```diff
@@ -6,36 +6,35 @@ import React from 'react';
...
-				<CompactCard>
-					<Button className="confirm-cancel-purchase-loading-placeholder__cancel-button" />
-				</CompactCard>
-			</LoadingPlaceholder>
-		);
-	}
-} );
+ConfirmCancelPurchaseLoadingPlaceholder.propTypes = {
+	purchaseId: React.PropTypes.string.isRequired,
+	selectedSite: React.PropTypes.object.isRequired
```
Comment (drewblaisdell) [inline client/me/purchases/confirm-cancel-purchase/loading-placeholder.jsx]: This component is also mounted with a false `selectedSite` if local storage is empty.

Comment (stephanethomas) [inline client/me/purchases/confirm-cancel-purchase/loading-placeholder.jsx]: That's a bug that existed before this change. What do you think of addressing that specific issue in a different pull request?

Comment (drewblaisdell) [inline client/me/purchases/confirm-cancel-purchase/loading-placeholder.jsx]: Sounds good. I didn't realize that before - I got the change in this file confused with the change in `propTypes` in `CancelPurchaseLoadingPlaceholder`.

File: client/me/purchases/cancel-private-registration/index.jsx [hunk 3]
```diff
@@ -12,14 +12,19 @@ import Button from 'components/button';
...
-import paths from '../paths';
 import Notice from 'components/notice';
+import paths from '../paths';
 import titles from 'me/purchases/titles';
-import { goToManagePurchase, isDataLoading, recordPageView } from '../utils';
 
 const CancelPrivateRegistration = React.createClass( {
+	propTypes: {
+		selectedPurchase: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.object.isRequired
```
Comment (drewblaisdell) [inline client/me/purchases/cancel-private-registration/index.jsx]: `selectedSite` can also be false here if local storage is empty.

Comment (drewblaisdell) [inline client/me/purchases/cancel-private-registration/index.jsx]: `selectedSite` can also be false here if local storage is empty.

File: client/me/purchases/manage-purchase/index.jsx [hunk 4]
```diff
@@ -67,14 +67,10 @@ function isDataLoading( props ) {
...
-		selectedPurchase: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.oneOfType( [
-			React.PropTypes.object,
-			React.PropTypes.bool,
-			React.PropTypes.undefined
-		] ),
 		destinationType: React.PropTypes.string,
+		hasLoadedSites: React.PropTypes.bool.isRequired,
+		selectedPurchase: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.object.isRequired,
```
Comment (drewblaisdell) [inline client/me/purchases/manage-purchase/index.jsx]: I believe the previous three types for `selectedSite` was accurate, for when the selected site is loading, present, or when `sites-list` has been fetched but the site in the URL isn't present.

**Comment by drewblaisdell** [comment]
Looks good. :+1: I had a couple notes about changes to `selectedSite` in `propTypes`.

**Comment by stephanethomas** [comment]
Thanks for your feedback Drew! I've addressed your comments and rebased.

This is ready for another review.

**Comment by scruffian** [comment]
LGTM

**Pull request merged**